### PR TITLE
Move Chef::OscUser back to Chef::User namespace and new user code to Chef::UserV1.

### PR DESCRIPTION
**Pull request by tylercloke**
Also, have Chef::User (formally Chef::OscUser) and the original Chef::ApiClient use API V0 to make requests.

Basically, grep / file rename changes. The only parts of this that need in-depth review are the following commits. Much smaller PR than the diff implies :)

https://github.com/chef/chef/commit/694a5335084a8692008dca2b698c60a35bdd1fed
https://github.com/chef/chef/commit/c14f6de738352579c5c751ead364d075adcbc66d
https://github.com/chef/chef/commit/3de16c56352a29c9d4688fea8b0b87efff6a1c95
https://github.com/chef/chef/commit/c54b68de500c9690f608f467aed40fabc21e9515
https://github.com/chef/chef/commit/1e606ac53cfaca6fa761511156801ee1a333b712
https://github.com/chef/chef/commit/d536641688eab73125a5dd3361f5301dc0490f80
https://github.com/chef/chef/commit/a071e84d52e81f6f1783094b2dd5bdd0e5e23b9f

Fixes https://github.com/chef/chef/issues/3628

TODO:
- [x] \(Lots of\) manual testing (help wanted).
- [x] Fixes open source chef server upgrade.
- [x] Changelog.
- [x] Rebase.

Ping @jaym @danielsdeleo @stevendanna cc/ client peeps

### CODE REVIEW

**Comment by jaym** [comment]
cc @chef/client-maintainers

**Comment by jaym** [comment]
/Users/jmundrawala/workspace/chef/lib/chef/user.rb:178:in `load': undefined method `get_rest' for #<Chef::ServerAPI:0x007fef3cd91ba0> (NoMethodError)

**Comment by jaym** [comment]
@tylercloke try `get` instead of `get_rest`

**Comment by jaym** [comment]
```
From: /Users/jmundrawala/workspace/chef-vault/lib/chef-vault/item_keys.rb @ line 37 ChefVault::ItemKeys#add:

    32: def add(chef_client, data_bag_shared_secret, type)
    33:   unless @raw_data.key?(type)
    34:     raise ChefVault::Exceptions::V1Format,
    35:           'cannot manage a v1 vault.  See UPGRADE.md for help'
    36:   end
 => 37:   require 'pry';binding.pry
    38:   public_key = OpenSSL::PKey::RSA.new chef_client.public_key
    39:   self[chef_client.name] =
    40:     Base64.encode64(public_key.public_encrypt(data_bag_shared_secret))
    41:
    42:   @raw_data[type] << chef_client.name unless @raw_data[type].include?(chef_client.name)
    43:   @raw_data[type]
    44: end

2.1.2 (data_bag_item[foo_keys]):0 > chef_client.class
=> Chef::ApiClient
2.1.2 (data_bag_item[foo_keys]):0 > chef_client
=> #<Chef::ApiClient:0x007fd72bdc6d38 @admin=false, @create_key=nil, @name="fundamentals", @private_key=nil, @public_key=nil, @validator=false>
```
When I get a `Chef::ApiClient`, `public_key` is `nil`

**Comment by danielsdeleo** [comment]
yeah, PSA: just use `get` `put` `post` `delete` everywhere. The `_rest` is redundant. :stars:

**Comment by tylercloke** [comment]
@jaym Yeah fixes to Chef Vault coming separately, thanks for noting them though.

**Comment by tylercloke** [comment]
^ Needs testing with Chef-Vault and verification from @jkeiser that Chef::ChefFS::FileSystem::ChefServerRootDir.chef_rest is safe to remove in favor of .rest. It was only being used in one place in the ChefFS code which was easy to update.

If we need to leave this interface due to external usage, that is fine though. I was just trying to clean up vestigal code. Please advise @jkeiser, @jaym, and @danielsdeleo.

**Comment by jaym** [comment]
If you'd like to remove an interface, please deprecate it. We can remove it in chef 13. `Chef::Log.depreaction` is the way to do that.

**Comment by jaym** [comment]
```
jmundrawala@seamundrawala01 ~/workspace/chef (●●)$ bundle exec knife vault create -VV foo foo '{}' --admins fundamentals --mode client                                                                                            [2.1.2]
INFO: Using configuration from /Users/jmundrawala/.chef/knife.rb
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::JSONToModelOutput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_request
DEBUG: Signing the request as jdmundrawala
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_request
DEBUG: Initiating GET to https://api.opscode.com/organizations/mundrawala-test/data/foo/foo_keys
DEBUG: ---- HTTP Request Header Data: ----
DEBUG: Accept: application/json
DEBUG: Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
DEBUG: X-OPS-SIGN: algorithm=sha1;version=1.0;
DEBUG: X-OPS-USERID: jdmundrawala
DEBUG: X-OPS-TIMESTAMP: 2015-07-02T16:56:33Z
DEBUG: X-OPS-CONTENT-HASH: 2jmj7l5rSw0yVb/vlWAYkK/YBwk=
DEBUG: X-OPS-AUTHORIZATION-1: HaIIjjVSh09ObDoCIElrRjOYNc99kSVKVU8HISFxKEIFv2h15Vzry9cE1fwh
DEBUG: X-OPS-AUTHORIZATION-2: 7O7SjEugxoT7+5in16ow2qjR3tNhYaDjKT4yLSQVwz4XyXEMtOjcu2Qw1Lj3
DEBUG: X-OPS-AUTHORIZATION-3: tjXqb8GU2xGbOlsgJFauMSjaqQ70Ged4iW5XWxumrk+0QR04nKJJO6+jOpU6
DEBUG: X-OPS-AUTHORIZATION-4: LpujqLFyV/x3e7NFz/KDGo0ZJoIBGzMERBumwcUYKSanajF2MIbX4fIpiMIF
DEBUG: X-OPS-AUTHORIZATION-5: mOEpErl36/gUwuKWtxQC6EFHxWCxxNEltt+TM9LePrBdYYoSTcqrey9e+nXu
DEBUG: X-OPS-AUTHORIZATION-6: j32EOFJHTvnVVB3YWqhejQnxCEzgoiZHPCtQKoubhw==
DEBUG: HOST: api.opscode.com:443
DEBUG: X-Ops-Server-API-Version: 1
DEBUG: X-REMOTE-REQUEST-ID: 5ae79d3d-b0b9-4adf-bee8-123e3f5eac53
DEBUG: ---- End HTTP Request Header Data ----
DEBUG: ---- HTTP Status and Header Data: ----
DEBUG: HTTP 1.1 404 Object Not Found
DEBUG: server: ngx_openresty
DEBUG: date: Thu, 02 Jul 2015 16:56:35 GMT
DEBUG: content-type: application/json
DEBUG: content-length: 65
DEBUG: connection: close
DEBUG: x-ops-server-api-version: {"min_version":"0","max_version":"1","request_version":"1","response_version":"1"}
DEBUG: x-ops-api-info: flavor=hec;version=12.0.0;oc_erchef=12.1.0
DEBUG: ---- End HTTP Status/Header Data ----
DEBUG: ---- HTTP Response Body ----
DEBUG: {"error":["Cannot load data bag item foo_keys for data bag foo"]}
DEBUG: ---- End HTTP Response Body -----
DEBUG: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_response
DEBUG: Content-Length validated correctly.
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONToModelOutput#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_response
INFO: HTTP Request Returned 404 Object Not Found: Cannot load data bag item foo_keys for data bag foo
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::JSONOutput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_request
DEBUG: Signing the request as jdmundrawala
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_request
DEBUG: Initiating GET to https://api.opscode.com/organizations/mundrawala-test/users/jdmundrawala
DEBUG: ---- HTTP Request Header Data: ----
DEBUG: Accept: application/json
DEBUG: Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
DEBUG: X-OPS-SIGN: algorithm=sha1;version=1.0;
DEBUG: X-OPS-USERID: jdmundrawala
DEBUG: X-OPS-TIMESTAMP: 2015-07-02T16:56:35Z
DEBUG: X-OPS-CONTENT-HASH: 2jmj7l5rSw0yVb/vlWAYkK/YBwk=
DEBUG: X-OPS-AUTHORIZATION-1: nFvc4Fzdh3CCPHAY3HK/CEnqr2Y0dpvAdwV0x6iy9/zH0XHRui588MF0kzNx
DEBUG: X-OPS-AUTHORIZATION-2: 6RyFZeC2UU5LUO1MWh6HJKjx0gsCtumWVoxPoxyXDHpS8Q7tc7MB/kQQSsnk
DEBUG: X-OPS-AUTHORIZATION-3: DjaIGWGRC5koAAn+QTP0Ik8uIi6Blipf9Sr2cZ5zoLwsQAeIoXgq7+gP4/74
DEBUG: X-OPS-AUTHORIZATION-4: deFUMVb0nfACp7NE01Uqx3vcNPbmrb4GLs3qzeA6F+Yc4xbQz+1216bIFVtd
DEBUG: X-OPS-AUTHORIZATION-5: zMAvSofNG5+bZXUCSodd3ZI/eCSwBYytlav3IrbWm8u38Q1ilJSt8nAngPiu
DEBUG: X-OPS-AUTHORIZATION-6: JIuIRluMidM2CctfzkN0+EgSm3YHOd0/2t2FrBqj9w==
DEBUG: HOST: api.opscode.com:443
DEBUG: X-Ops-Server-API-Version: 0
DEBUG: X-REMOTE-REQUEST-ID: 5ae79d3d-b0b9-4adf-bee8-123e3f5eac53
DEBUG: ---- End HTTP Request Header Data ----
DEBUG: ---- HTTP Status and Header Data: ----
DEBUG: HTTP 1.1 200 OK
DEBUG: server: ngx_openresty
DEBUG: date: Thu, 02 Jul 2015 16:56:35 GMT
DEBUG: content-type: application/json
DEBUG: content-length: 607
DEBUG: connection: close
DEBUG: x-ops-server-api-version: {"min_version":"0","max_version":"1","request_version":"0","response_version":"0"}
DEBUG: x-ops-api-info: flavor=hec;version=12.0.0;oc_erchef=12.1.0
DEBUG: ---- End HTTP Status/Header Data ----
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONOutput#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::JSONOutput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_request
DEBUG: Signing the request as jdmundrawala
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_request
DEBUG: Initiating GET to https://api.opscode.com/organizations/mundrawala-test/users/fundamentals
DEBUG: ---- HTTP Request Header Data: ----
DEBUG: Accept: application/json
DEBUG: Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
DEBUG: X-OPS-SIGN: algorithm=sha1;version=1.0;
DEBUG: X-OPS-USERID: jdmundrawala
DEBUG: X-OPS-TIMESTAMP: 2015-07-02T16:56:35Z
DEBUG: X-OPS-CONTENT-HASH: 2jmj7l5rSw0yVb/vlWAYkK/YBwk=
DEBUG: X-OPS-AUTHORIZATION-1: cA+GYxWfxgAMF4frqhZFUHd2cCJZIS6aVpGUgFxVZmbyZSnAYwOEvUGgB4vG
DEBUG: X-OPS-AUTHORIZATION-2: rbSFO4W/f/LfL1u7/QrES5BdKo/BDfsKfbyVEUHjFlhhIA6croUiq0R94f86
DEBUG: X-OPS-AUTHORIZATION-3: KjktKOY21/DCATYY9GQnkc3Wy9GFko9ooHFhc71o7QnCHPk+Cl217Xz9jpNT
DEBUG: X-OPS-AUTHORIZATION-4: 56w6Fo4vdQ1Es0onA3pTjy619nX38F6s3ICyGHWyZzGwoqoPU8ERB8eL2WXs
DEBUG: X-OPS-AUTHORIZATION-5: +BPYFxlIbB25HfJXxGKZo9a3/gUrVf8/MZdSDbWdl5FZ0FtQd54dbPDNjcmL
DEBUG: X-OPS-AUTHORIZATION-6: L1N51jj6b2u7LaeTctKkIokmHZvc+QgbHXrZGGp5KA==
DEBUG: HOST: api.opscode.com:443
DEBUG: X-Ops-Server-API-Version: 0
DEBUG: X-REMOTE-REQUEST-ID: 5ae79d3d-b0b9-4adf-bee8-123e3f5eac53
DEBUG: ---- End HTTP Request Header Data ----
DEBUG: ---- HTTP Status and Header Data: ----
DEBUG: HTTP 1.1 404 Object Not Found
DEBUG: server: ngx_openresty
DEBUG: date: Thu, 02 Jul 2015 16:56:36 GMT
DEBUG: content-length: 44
DEBUG: connection: close
DEBUG: x-ops-server-api-version: {"min_version":"0","max_version":"1","request_version":"0","response_version":"0"}
DEBUG: x-ops-api-info: flavor=hec;version=12.0.0;oc_erchef=12.1.0
DEBUG: ---- End HTTP Status/Header Data ----
DEBUG: ---- HTTP Response Body ----
DEBUG: {"error":"Could not find user fundamentals"}
DEBUG: ---- End HTTP Response Body -----
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONOutput#handle_response
DEBUG: Expected JSON response, but got content-type ''
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_response
INFO: HTTP Request Returned 404 Object Not Found: error
WARNING: fundamentals not found in users, trying clients.
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::JSONToModelOutput#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_request
DEBUG: Signing the request as jdmundrawala
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_request
DEBUG: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_request
DEBUG: Initiating GET to https://api.opscode.com/organizations/mundrawala-test/clients/fundamentals
DEBUG: ---- HTTP Request Header Data: ----
DEBUG: Accept: application/json
DEBUG: Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
DEBUG: X-OPS-SIGN: algorithm=sha1;version=1.0;
DEBUG: X-OPS-USERID: jdmundrawala
DEBUG: X-OPS-TIMESTAMP: 2015-07-02T16:56:36Z
DEBUG: X-OPS-CONTENT-HASH: 2jmj7l5rSw0yVb/vlWAYkK/YBwk=
DEBUG: X-OPS-AUTHORIZATION-1: njoSq4z0a7X9lH1UFV5tYgWUSxYngvwfetk+HOjacpsWsK/rky+IYhrWuNKw
DEBUG: X-OPS-AUTHORIZATION-2: uUuxM/t5QRfxJB4ZaEXDftZREAK+xukUHwOr/NbaCla+efgBtwelGMWI9h4I
DEBUG: X-OPS-AUTHORIZATION-3: Jq+7ZyCDAHaikbSxynHlHmxaBN9yNGThBXA4tdHLZUh0owtJKVatC+9Z0RMz
DEBUG: X-OPS-AUTHORIZATION-4: wr7nIwWQ+LHt6DY4huaKHQUxbK5dkYW+OF6r8n/BGfy/YQia4tMcJ7iTyHeK
DEBUG: X-OPS-AUTHORIZATION-5: cRKnDGJC3ZqB7capbgdSLIifY8JxHzDKw3yQXMYkFpJ4GJPiRHqXyff1LOrZ
DEBUG: X-OPS-AUTHORIZATION-6: Mz9MFVNR23BCMQ7Lv2YcdPQQhtHi8Zb31gE8xK0b0Q==
DEBUG: HOST: api.opscode.com:443
DEBUG: X-Ops-Server-API-Version: 1
DEBUG: X-REMOTE-REQUEST-ID: 5ae79d3d-b0b9-4adf-bee8-123e3f5eac53
DEBUG: ---- End HTTP Request Header Data ----
DEBUG: ---- HTTP Status and Header Data: ----
DEBUG: HTTP 1.1 200 OK
DEBUG: server: ngx_openresty
DEBUG: date: Thu, 02 Jul 2015 16:56:37 GMT
DEBUG: content-type: application/json
DEBUG: content-length: 149
DEBUG: connection: close
DEBUG: x-ops-server-api-version: {"min_version":"0","max_version":"1","request_version":"1","response_version":"1"}
DEBUG: x-ops-api-info: flavor=hec;version=12.0.0;oc_erchef=12.1.0
DEBUG: ---- End HTTP Status/Header Data ----
DEBUG: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_response
DEBUG: Content-Length validated correctly.
DEBUG: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Authenticator#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::Decompressor#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::CookieManager#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONToModelOutput#handle_response
DEBUG: Chef::HTTP calling Chef::HTTP::JSONInput#handle_response
/Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef-vault/item_keys.rb:33:in `initialize': no implicit conversion of nil into String (TypeError)
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef-vault/item_keys.rb:33:in `new'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef-vault/item_keys.rb:33:in `add'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef-vault/item.rb:121:in `block in admins'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef-vault/item.rb:117:in `each'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef-vault/item.rb:117:in `admins'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef/knife/vault_create.rb:84:in `rescue in run'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/chef-vault-2.6.1/lib/chef/knife/vault_create.rb:57:in `run'
	from /Users/jmundrawala/workspace/chef/lib/chef/knife.rb:425:in `block in run_with_pretty_exceptions'
	from /Users/jmundrawala/workspace/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
	from /Users/jmundrawala/workspace/chef/lib/chef/knife.rb:424:in `run_with_pretty_exceptions'
	from /Users/jmundrawala/workspace/chef/lib/chef/knife.rb:215:in `run'
	from /Users/jmundrawala/workspace/chef/lib/chef/application/knife.rb:142:in `run'
	from /Users/jmundrawala/workspace/chef/bin/knife:25:in `<top (required)>'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/bin/knife:23:in `load'
	from /Users/jmundrawala/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/bin/knife:23:in `<main>'
```

**Comment by jaym** [comment]
There is a `Chef::ApiClient` object coming back with no public key

**Comment by jaym** [comment]
I don't want to block on this, but it's a little concerning:
```
jmundrawala@seamundrawala01 ~/workspace/chef (●)$ git checkout origin/tc/fix-chef-user           [2.1.2]
Previous HEAD position was 283b8d2... Merge pull request #3580 from chef/ksubrama/release-notes
HEAD is now at 3c19814... Remove all the stupid json magic from Chef::ApiClientV1.
jmundrawala@seamundrawala01 ~/workspace/chef (●)$ bundle exec knife client show fundamentals     [2.1.2]
admin:     false
chef_type: client
name:      fundamentals
validator: false

jmundrawala@seamundrawala01 ~/workspace/chef (●)$ git checkout 12.3.0                            [2.1.2]
Previous HEAD position was 3c19814... Remove all the stupid json magic from Chef::ApiClientV1.
HEAD is now at cb8eb59... Release 12.3.0
jmundrawala@seamundrawala01 ~/workspace/chef (●)$ bundle exec knife client show fundamentals     [2.1.2]
admin:      false
chef_type:  client
json_class: Chef::ApiClient
name:       fundamentals
public_key: -----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEApF+KpWeKYFRQ2hGir9w3
frJ6mLWK4noMCU2cJVSoWd68ZMUphjZZOiwTkA31cMwwKYy7k7nu6jbq1lw2Fp9h
nYuUwGo3znuRydmttGyvc5ifNj6kMOjavUFFf3hTf8DUFq0lmjzbYmxVn7vABr8Z
vUCMQIVHM4VL9btZt6fEr8/RG5UJB4Aswp9AVLAhh/e7/cd3NHVCRbVzuiXr7cYw
KJHXtRmeco9xG9GRyHcWw0GlPdk6Keut4EVAXD70ZLjgyaBGDX9QBPgej0ZxKEm0
gcyaykSIefv3yXi331AriJZPrT4CTTqP9aB6dNPIiDtJ/5bZHwBOy2u4jSL9oiD4
GQIDAQAB
-----END PUBLIC KEY-----

validator:  false
```

Could break people who are parsing the output of our knife commands. Don't know how common that is

File: lib/chef/chef_fs/file_system/cookbooks_dir.rb [hunk 1]
```diff
@@ -106,7 +106,7 @@ def upload_versioned_cookbook(other, options)
             cookbook_to_upload.freeze_version if options[:freeze]
 
             # Instantiate a new uploader based on the proxy loader
-            uploader = Chef::CookbookUploader.new(cookbook_to_upload, :force => options[:force], :rest => root.chef_rest)
+            uploader = Chef::CookbookUploader.new(cookbook_to_upload, :force => options[:force], :rest => root.rest)
```
Comment (jaym) [inline lib/chef/chef_fs/file_system/cookbooks_dir.rb]: What is `root.rest`
It seems like whatever that is it should probably map `chef_rest` to `rest`

Comment (tylercloke) [inline lib/chef/chef_fs/file_system/cookbooks_dir.rb]: It should but its breaking tests. I'm gonna leave it as `chef_rest` but we will be deprecating it in 13.

Comment (jaym) [inline lib/chef/chef_fs/file_system/cookbooks_dir.rb]: that is also acceptable

**Comment by jaym** [comment]
:+1:

**Comment by jkeiser** [comment]
Trying to understand what's pending: is this ready to merge?

**Comment by tylercloke** [comment]
Yes, running it through the chef-server to see if it fixes the issue I was seeing there first is all.

**Comment by tylercloke** [comment]
https://github.com/chef/chef-server/issues/371

**Comment by danielsdeleo** [comment]
:+1:

**Pull request merged**